### DISCUSSION TITLE
Two Observations -- Are they related?

### DISCUSSION THREAD

**Discussion by FA-Bubba** [SamDel/ChromeCast-Desktop-Audio-Streamer#127] [Q&A] [answered]
Many of my speakers would have frequent audio drop-outs, sometimes displaying "Connect Error"; other times "Check Firewall" or just the 'Orange color' with no Label.  In the past, it was explained to me that the Group "Leader" was likely causing this, so about 1 month ago, I quit using groups, and connected each speaker individually via the DAS Device Box.

The frequent drop-outs ceased, and any occasional drop-outs quickly reconnected on their own.

Early this morning, my Xfinity Gateway went off-line, so ALL clients on my LAN lost internet access.  However, ALL of my DAS devices continued to Cast.  (I have previously mentioned that I have a Mesh network that gets it internet via EtherNet cable from the Xfinity Gateway, but the Mesh network Wi-Fi works independently from the Xfinity Wi-Fi.)

The Gateway was down for over two hours, and the EtherNet cable connection did not have a signal, so ALL Clients, including ALL of my speakers lost Internet access.  I had previously been informed that the Google devices (Minis; Hub; ChromeCast Audio, etc) needed an Internet connection to function, but none of them dropped-out while the Internet connection was lost...

Has something changed, or did they stay up because I wasn't using "Groups" in DAS anymore?   ...or is there another explanation?

**Comment by SamDel** [comment, marked as answer]
We did this test in 2020: https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/issues/87#issuecomment-670986668

> Test results:
> 
> - When a device is playing and you unplug the (main) internet cable, DAS continues playing.
> - When a device is idle and you unplug the (main) internet cable, DAS can start streaming to the device.
> - When a device is turned on when no internet is available it refuses connections ("Connect Error") until internet is available.

I don't think anything has changed since then, but today you didn't reboot a device when the internet connection was down.

**Reply by FA-Bubba** [reply]
Sorry, I failed to look up your test from 2020...  I understand the need for Internet when a Google Device is initially turned on/plugged in.

As for the improved stability of my DAS Devices over the past month, I probably won't use Audio Groups anymore. (unless the "Leader Issue" can be resolved.

Thank you for your help!

**Reply by SamDel** [reply]
> unless the "Leader Issue" can be resolved

You may have tried it before but the only thing I can think of (in this app) is to set the 'Device Buffer (in seconds)' to 20 and see if that works better.

**Reply by FA-Bubba** [reply]
I finally got around to making this change... My DAS version is 3.9.0.0, and I set the buffer to 20; however, when I clicked "Reset Settings", it defaulted back to 4 -- is that the way it is supposed to work, or is there a way for me to change the default to 20?

(Sorry if this a dupe; I closed the page before I sent this, so I am resending).

**Reply by SamDel** [reply]
It's intended to work that way, the buffer defaults to 4 seconds ([code](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/blob/5dbc39a34f542c0587be09750fa3764f48ca7c3d/Source/ChromeCast.Desktop.AudioStreamer/Application/ApplicationLogic.cs#L434)). 
There's no way of changing the default (besides changing the code).

**Reply by FA-Bubba** [reply]
OK...  FYI, the drop-outs were worse, more often, at 20 seconds, So I have it back to 4.

Thanks!
